### ISSUE TITLE
Inline assignment inside while statement

### ISSUE THREAD

**Issue by hishamco** [dotnet/vblang#175]
Moved from https://visualstudio.uservoice.com/forums/121579-visual-studio-ide/suggestions/5804781-inline-assignment-inside-while-statement?tracking_code=b8ad6215b353994f320715a0d9ee7569

I think VB developers miss the inline assignment inside while clause feature, which is already exist in c#
its great if we can find the equivalent for this in VB

```csharp
while ((bytesRead = stream.Read(buffer, 0, buffer.Length)) > 0) 
{ 
// Use the data you've read 
}
```

will be

```vb
While (bytesRead = stream.Read(buffer, 0, buffer.Length)) > 0 
' Use the data you've read 
End While
```

I know the VB uses = as equality operator inside while, but while VB uses the same operator for assignment, perhaps we can find a proper solution for this

**Comment by rskar-git** [comment]
Inventing an Assign function like so works fine:

    Function Assign(Of T)(ByRef x As T, result As T) As T 
        x = result 
        Return result 
    End Function

    Do While Assign(bytesRead, stream.Read(buffer, 0, buffer.Length)) > 0 
        ' Use the data you've read 
    Loop

Or maybe an extension instead:

    <Extension>
    Function [Let](Of T)(ByRef x As T, result As T) As T
        x = result
        Return result
    End Function

    Do While bytesRead.Let(stream.Read(buffer, 0, buffer.Length)) > 0 
        ' Use the data you've read 
    Loop

I do admit, though, sometimes I toy with this syntax idea:

    Do While (Let bytesRead = stream.Read(buffer, 0, buffer.Length) > 0 
        ' Use the data you've read 
    Loop

So the idea is if a `Let` happens immediately after `(`, then you can have an l-value, then `=` (or `+=` or `-=` or `*=`, etc.), expression, closing `)`. Kind of gives `Let` another reason for being (and hopefully won't interfere with the `Let` of Linq).

**Comment by ericmutta** [comment]
I agree with this, inline assignment AND declaration are very handy, and in MANY areas, not just the While loop...see #159 for more.

**Comment by xieguigang** [comment]
This can be done by a ``=`` operator overloads helper, like [Microsoft.VisualBasic.Language.Value(Of T)](https://github.com/xieguigang/sciBASIC/blame/a452b1d4356d2ab0d39e6a6f71ee807320a41ede/Microsoft.VisualBasic.Architecture.Framework/Language/Value/Value.vb#L202):

Usage:

```vbnet
Imports Microsoft.VisualBasic.Language

Dim s As Value(Of String) = ""

Do While Not (s = blablabla) Is Nothing
   ' Do other staff
Loop
```
